### DISCUSSION TITLE
Need examples of how to set config

### DISCUSSION THREAD

**Discussion by alecazam** [ccache/ccache#1468] [Support]
Don't seem to be any samples of setting config from the command line.  I need to set sloppiness for pch use (at least according to the docs).  I don't want to have a file, and want to set the config from the command line.  It doesn't seem to be working.   

Is is this?  Or am I setting CCACHE_SLOPPINESS=pch_defines,time_macros
sloppiness=pch_defines,time_macros

macOS ccache states this form is possible from command line (doesn't work from Make)
ccache --set-config name=value compiler [compiler args]

Win states that it needs an = in the usage  (doesn't work from Make)
cache --set-config=name=value compiler [compiler args]

This seems to work from macOS Make, but fails on Win (thinks sloppiness is a compiler)
ccache name=value compiler [compiler args]

**Comment by alecazam** [comment]
This is the usage from macOS.  There's not a version where you set ccache options and set the compiler and compiler options.  Are these settings sticky to the next call?

```
    ccache [ccache options]
    ccache [KEY=VALUE ...] compiler [compiler options]
    compiler [compiler options]
```

**Comment by alecazam** [comment]
C:/Foo/ccache sloppiness=pch_defines,time_macros ....

8>ccache : error : Could not find compiler "sloppiness=pch_defines,time_macros" in PATH

**Comment by alecazam** [comment]
Seemed to be a problem with an older ccache (3.7.1) on Win vs. (4.9.1) on macOS.   Updating seems to have fixed.   Still think posting samples of config would be useful.    Also think it's not helpful to list out command line args, and then only take name=value pairs in the version that I can call the compiler and compiler args on.

**Comment by jrosdahl** [comment]
@alecazam wrote:

> Don't seem to be any samples of setting config from the command line.

There is an example here: https://ccache.dev/manual/4.10.html#_configuration

The `ccache [KEY=VALUE …​] compiler [compiler options]` form is for specifying configuration for a single ccache invocation. It is not retained.

The `ccache --set-config KEY=VALUE` form (which can also be written `ccache --set-config=KEY=VALUE`) is for writing to the configuration file. It does not specify a value for a single ccache invocation, it's just an alternative to editing the configuration file manually.

> Seemed to be a problem with an older ccache (3.7.1)

Right, the `ccache [KEY=VALUE …​] compiler [compiler options]` form` was added in ccache 4.8.

> Also think it's not helpful to list out command line args, and then only take name=value pairs in the version that I can call the compiler and compiler args on.

Ccache supports exactly the forms that are documented. I don't understand what you think isn't helpful.
